### DISCUSSION TITLE
[QUESTION] Is there any APM that supports swoole?

### DISCUSSION THREAD

**Discussion by Ananiaslitz** [swoole/swoole-src#5096] [Q&A] [answered]
Is there any APMs that officially supports Swoole?

**Comment by leocavalcante** [comment, marked as answer]
The [Apache SkyWalking](https://skywalking.apache.org/) PHP agent seems to have an eye at Swoole ecosystem: https://github.com/apache/skywalking-php/tree/master#support-list

However, I highly recommend the use of OpenTelemetry as it is a standard that grows each time more and more in the industry of APMs, also making you project vendor-free/neutral.

With `hyperf/tracer` and `hyperf/metric` (if you are using Hyperf, for example) you are able to send open formats like Zipkin, Jaeger, Prometheus and Statsd to an OpenTelemetry Collector and export them to any APM platform you like: New Relic, Dynatrace, DataDog, Honeycomb etc or your own setup of Jaeger UI and Prometheus+Grafana.
